pkg/icingadb/v1/customvar.go

```diff
@@ -117,21 +116,14 @@ func flattenCustomvars(ctx context.Context, g *errgroup.Group, cvs <-chan contra
 					flattened := flatten.Flatten(value, customvar.Name)
 
 					for flatname, flatvalue := range flattened {
-						var fv string
-						if flatvalue == nil {
-							fv = "null"
-						} else {
-							fv = fmt.Sprintf("%v", flatvalue)
-						}
-
 						select {
 						case flatCustomvars <- &CustomvarFlat{
 							CustomvarMeta: CustomvarMeta{
 								EntityWithoutChecksum: EntityWithoutChecksum{
 									IdMeta: IdMeta{
 										// TODO(el): Schema comment is wrong.
 										// Without customvar.Id we would produce duplicate keys here.
-										Id: utils.Checksum(objectpacker.MustPackSlice(customvar.EnvironmentId, customvar.Id, flatname, flatvalue)),
+										Id: utils.Checksum(objectpacker.MustPackSlice(customvar.EnvironmentId, customvar.Id, flatname, flatvalue.String)),
```

Comment:
> I think it will at least rewrite all non-string values (for those, the ID should stay identical).

I think the only way to achieve this is to revert my changes.

Reply:
That's not what I was trying to say with that: If `flatvalue` in the old code was of type `string`, the ID generated by the new code should be the same. Only those for values of other types would change.

Now it's up to us to assess the effects of that and depending on the result, those change may or may not be revert.

Reply:
`customvar.id` should not be affected by the change, just `customvar_flat.id`. And this should really be referenced nowhere, so it might be fine to change this if we say it's okay to potentially rewrite a good part of the `customvar_flat` table on the first sync after the upgrade.